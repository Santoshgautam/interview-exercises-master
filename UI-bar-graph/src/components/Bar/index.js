import React from "react";

const Bar = ({ data, label, max }) => {
  return (
    <div className="bar" style={{ height: `${(data / max) * 100}%` }}>
      <span className="bar-value">{data}</span>
      <span className="bar-label">{label}</span>
    </div>
  );
};

export default Bar;
