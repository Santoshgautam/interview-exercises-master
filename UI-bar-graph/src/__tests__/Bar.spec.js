import Bar from "../components/Bar";

describe("Bar Component", () => {
  let wrapper, consoleError;

  beforeEach(() => {
    wrapper = shallow(<Bar />);
    consoleError = console.error;
    console.error = jest.spyOn(console, "error");
  });

  afterEach(() => {
    console.error = consoleError;
  });

  it("renders without crashing", () => {
    expect(wrapper.exists()).toBe(true);
  });

  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("displays correct value", () => {
    const data = 45;
    wrapper = shallow(<Bar data={data} />);
    expect(wrapper.find(".bar-value").text()).toEqual(String(data));
  });

  it("displays correct label", () => {
    const label = "Some label";
    wrapper = shallow(<Bar label={label} />);
    expect(wrapper.find(".bar-label").text()).toEqual(label);
  });
});
