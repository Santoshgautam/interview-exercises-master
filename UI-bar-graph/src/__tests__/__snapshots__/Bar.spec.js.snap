// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar Component matches the snapshot 1`] = `ShallowWrapper {}`;
